src/python/pants/backend/java/subsystems/javac.py

```diff
@@ -16,6 +16,7 @@ class JavacSubsystem(Subsystem):
     name = "javac"
     help = "The javac Java source compiler."
 
+    skip = SkipOption("check")
```

Comment:
This option is weird: `--javac-skip`. Maybe it should be more explicit like `--javac-check-skip`? You're not skipping Javac in general, like Javac will still run for tests!

Reply:
> You're not skipping Javac in general, like Javac will still run for tests!

I actually think that is a bug we just haven't fixed yet. The way I see it there should be multiple levels to skipping/omission for ALL goals where >1 tool is involved.
- `--only` on the goal
- `skip` on the tool subsystem (in which case skips that tool from all relevant goals, including `test` :stuck_out_tongue: )
- `skip_<tool_name>` on the goal. Like `skip` on the tool subsystem, but registered via a plugin option on just the goal
- (In the case of `lint` `--skip-formatters`)

Reply:
So in this case I'd advocate for `check.skip_javac`

Reply:
(I actually think that is a bug we just haven't fixed yet.) I can also fix these if that's the path forward.

Reply:
Yeah, I like `--check-skip-javac` more. But in the meantime to use doing that, how about going with `--javac-check-skip`?